tests/unit/Hooks/ImageHooksTest.php

```diff
@@ -202,10 +142,61 @@ protected function mockJpgFileMethods($node, $path) {
 	 * @param $path
 	 * @param $dimensions
 	 */
-	protected function mockGetImageSize($mockClass, $path, $dimensions) {
-		$mockClass->expects($this->once())
+	//TODO: This is still not correctly implemented
+	protected function mockGetImageSize($path, $dimensions) {
+		$this->imagehooks->expects($this->once())
 			->method('getimagesize')
 			->with($path)
 			->willReturn($dimensions);
 	}
+
+	/**
+	 * @return object|\PHPUnit_Framework_MockObject_MockObject
+	 */
+	private function mockImageDimensionMapper() {
+		/*return new ImageDimensionMapper(
+			$this->db
+		);*/
+
+		return $this->getMockBuilder('OCA\MediaMetadata\Services\ImageDimensionMapper')
+			->setConstructorArgs($this->db)
+			->getMock();
+
+		/*
+		 * Another Method
+		 * $app = new \OCA\MediaMetadata\AppInfo\Application();
+		 * $container = $app->getContainer();
+		 *
+		 * return new ImageDimensionMapper(
+		 *     $container->query('ServerContainer')->getDb()
+		 * );
+		 */
+	}
+
+	/**
+	 * @return object|\PHPUnit_Framework_MockObject_MockObject
+	 */
+	private function mockImageDimension() {
+		//return new ImageDimension();
+
+		return $this->getMockBuilder('OCA\MediaMetadata\Services\ImageDimension')
+			->disableOriginalConstructor()
+			->getMock();
+	}
+
+	/**
+	 * @param array $mockedMethods
+	 * @return object|ImageHooks|\PHPUnit_Framework_MockObject_MockObject
+	 */
+	private function mockPostCreate() {
```

Comment:
Yes, Okay. Actually, you had told to change the name of getImageHooks to mockPostCreate, so I had changed at that time but after that I changed the contents of this function as we do not need to mock post create in this test.  I will change it back to getImageHooks.